src/Controls/src/SourceGen/NodeSGExtensions.cs

```diff
@@ -0,0 +1,661 @@
+using System;
+using System.Xml;
+
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.Linq;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Diagnostics;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+using Microsoft.Maui.Controls.Xaml;
+
+namespace Microsoft.Maui.Controls.SourceGen;
+
+using static LocationHelpers;
+static class NodeSGExtensions
+{
+	public delegate string ConverterDelegate(string value, BaseNode node, ITypeSymbol toType, SourceGenContext context, LocalVariable? parentVar = null);
+
+	public delegate string ProvideValueDelegate(IElementNode markupNode, SourceGenContext context, out ITypeSymbol? returnType);
+
+	static Dictionary<ITypeSymbol, (ConverterDelegate, ITypeSymbol)>? KnownSGTypeConverters;
+
+	static Dictionary<ITypeSymbol, (ConverterDelegate, ITypeSymbol)> GetKnownSGTypeConverters(SourceGenContext context)
+		=> KnownSGTypeConverters ??= new(SymbolEqualityComparer.Default)
+	{
+		{ context.Compilation.GetTypeByMetadataName("Microsoft.Maui.Graphics.Converters.RectTypeConverter")!, (KnownTypeConverters.ConvertRect, context.Compilation.GetTypeByMetadataName("Microsoft.Maui.Graphics.Rect")!) },
```

Comment:
This used `global::` a lot in other source generators. Do any of the names in this file also need that?

I could make a type named `namespace Microsoft.Maui; class Graphics { }` would that break?